lib/sepa_king/message.rb

```diff
@@ -70,7 +70,7 @@ def schema_compatible?(schema_name)
 
     # Unique identifer for the whole message
     def message_identification
-      @message_identification ||= "SEPA-KING/#{Time.now.to_i}"
+      @message_identification ||= "SEPA-KING/#{SecureRandom.hex[0..22]}"
```

Comment:
I know I'm late to the party, but this still might cause some collisions. 

`SecureRandom.hex` generates a 32 character long string, from which you just get first 22 characters.

If you want truly random 22 characters, you ought to use `SecureRandom.hex(11)` as per [Ruby docs](http://ruby-doc.org/stdlib-2.1.2/libdoc/securerandom/rdoc/SecureRandom.html#method-c-hex)